### PR TITLE
Add end-to-end test for invalid contract start character 0xEF

### PR DESCRIPTION
**Pull request by facuMH**
This PR adds an end-to-end test for invalid contracts whose code starts with `0xEF` ([EIP-3541](https://eips.ethereum.org/EIPS/eip-3541))

The test added tests this from two different mechanisms:
- deploys a smart contract that provides 4 functions which attempt to create an account with valid/invalid code (where invalid is code that starts with byte `0xEF`) using the instructions `CREATE` or `CREATE2`
- creates two custom transactions with the provided valid/invalid codes and attempts to run them.

For all cases with invalid codes, the receipt is checked for `types.ReceiptStatusFailed`.
For all cases  with valid codes, the receipt is checked for `types.ReceiptStatusSuccessful`.

This PR also makes use of `require` library for error and results verification. (previously used in https://github.com/Fantom-foundation/Sonic/blob/develop/config/config_custom_test.go#L51)

This PR is part of https://github.com/Fantom-foundation/sonic-admin/issues/19

### CODE REVIEW

**Review by HerbertJordan — DISMISSED** [review]
Looks good, thanks!

File: tests/contracts/invalidstart/invalidstart.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    function create2ContractWithInvalidCode() public {
+        create2OrRevert(invalidBytecode);
+    }
+
+    function create2ContractWithValidCode() public {
+        create2OrRevert(validBytecode);
+    }
+
+    function create2OrRevert(bytes memory bytecode) public {
```
Comment (LuisPH3) [inline tests/contracts/invalidstart/invalidstart.sol:33]: nit: private

File: tests/contracts/invalidstart/invalidstart.sol [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    bytes public validBytecode = hex"60fe60005360016000f3";
+    function createContractWithInvalidCode() public {
+        createOrRevert(invalidBytecode);
+    }
+
+    function createContractWithValidCode() public {
+        createOrRevert(validBytecode);
+    }
+
+    function createOrRevert(bytes memory bytecode) public {
```
Comment (LuisPH3) [inline tests/contracts/invalidstart/invalidstart.sol:16]: nit: private

File: tests/invalid_start_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	r.NoError(err, "failed to get nonce:")
+
+	price, err := client.SuggestGasPrice(context.Background())
+	r.NoError(err, "failed to get gas price:")
+	// ---------
+
+	transaction, err := types.SignTx(types.NewTx(&types.AccessListTx{
+		ChainID:  chainId,
+		Gas:      500_000, // some gas that is big enough to run the code
+		GasPrice: price,
```
Comment (LuisPH3) [inline tests/invalid_start_test.go:85]: Not sure if we need gasPrice, I think that when it is  null it gets deduced.

File: tests/invalid_start_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		To:       nil,
+		Value:    big.NewInt(0),
```
Comment (LuisPH3) [inline tests/invalid_start_test.go:87]: no need for value in this TX

**Review by LuisPH3 — APPROVED** [review]

File: tests/invalid_start_test.go (hunk 3 above)
Comment (facuMH) [reply tests/invalid_start_test.go:85]: without a price a `transaction underpriced` error occurs.

Comment (HerbertJordan) [reply tests/invalid_start_test.go:84]: I do not think there is an auto-deduction of the price in this context. The `nil` value is likely treated like a zero.

**Review by HerbertJordan — APPROVED** [review]

**Review by LuisPH3 — CHANGES_REQUESTED** [review]
Please refactor to use one single network

**Review by LuisPH3 — APPROVED** [review]

**Pull request merged**